kratos/tests/cpp_tests/utilities/test_divide_triangle_3d_3.cpp

```diff
@@ -269,5 +266,33 @@ KRATOS_TEST_CASE_IN_SUITE(DivideGeometryTriangle3D3, KratosCoreFastSuite)
     KRATOS_CHECK_VECTOR_EQUAL((*neg_ext_faces[2])[1],neg_ext_faces_2_node_1_coordinates);
 
 }
+
+KRATOS_TEST_CASE_IN_SUITE(DivideGeometryTriangle3D3ZeroNode, KratosCoreFastSuite)
+{
+    Model current_model;
+
+    // Generate a model part with the previous
+    ModelPart& base_model_part = current_model.CreateModelPart("Triangle");
+
+    // Fill the model part geometry data
+    base_model_part.CreateNewNode(1, 0.3,  0.0, 0.6);
+    base_model_part.CreateNewNode(2, 0.3,  0.0, 0.9);
+    base_model_part.CreateNewNode(3, 0.3, -0.3, 0.3);
+    Properties::Pointer p_properties(new Properties(0));
+    auto p_elem = base_model_part.CreateNewElement("Element3D3N", 1, {1, 2, 3}, p_properties);
+
+    Vector nodal_distances = ZeroVector(3);
+    // cut at z = 0.6, across node 0
+    nodal_distances[0] = 0.0;
+    nodal_distances[1] = 0.3;
+    nodal_distances[2] = -0.3;
+
+    auto divider = DivideTriangle3D3<Node>(p_elem->GetGeometry(), nodal_distances);
+    divider.GenerateDivision();
+
+    // Should split a single edge and return two triangles
+    KRATOS_CHECK_EQUAL(divider.GetPositiveSubdivisions().size(), 1);
+    KRATOS_CHECK_EQUAL(divider.GetNegativeSubdivisions().size(), 1);
```

Comment:
I think this test must be added in the `test_divide_triangle_2d_3.cpp` instead of this one.